Hi @jaeeyoungkim , the changes and E2E tests are looking good. However, I'm seeing this failure with 'make test-main':

# github.com/NVIDIA/dcgm-exporter/internal/pkg/transformation [github.com/NVIDIA/dcgm-exporter/internal/pkg/transformation.test]
internal/pkg/transformation/kubernetes_test.go:400:22: podMapper.updateCache undefined (type *PodMapper has no field or method updateCache)
internal/pkg/transformation/kubernetes_test.go:594:19: podMapper.updateCache undefined (type *PodMapper has no field or method updateCache)
internal/pkg/transformation/kubernetes_test.go:824:19: podMapper.updateCache undefined (type *PodMapper has no field or method updateCache)
internal/pkg/transformation/kubernetes_test.go:939:19: podMapper.updateCache undefined (type *PodMapper has no field or method updateCache)

Can you try to reproduce and fix?

--- a/internal/pkg/transformation/kubernetes_test.go
+++ b/internal/pkg/transformation/kubernetes_test.go
@@ -396,11 +396,7 @@ func TestProcessPodMapper_WithD_Different_Format_Of_DeviceID(t *testing.T) {
 				mockSystemInfo.EXPECT().GPUCount().Return(uint(1)).AnyTimes()
 				mockSystemInfo.EXPECT().GPU(uint(0)).Return(mockGPU).AnyTimes()
 
-				// Update cache manually to populate deviceToPod mapping
-				err := podMapper.updateCache(mockSystemInfo)
-				require.NoError(t, err)
-
-				err = podMapper.Process(metrics, mockSystemInfo)
+				err := podMapper.Process(metrics, mockSystemInfo)
 				require.NoError(t, err)
 				assert.Len(t, metrics, 1)
 
@@ -590,11 +586,7 @@ func TestProcessPodMapper_WithLabels(t *testing.T) {
 	}
 
 	// Process metrics
-	// Update cache manually to populate deviceToPod mapping
-	err := podMapper.updateCache(mockDeviceInfo)
-	require.NoError(t, err)
-
-	err = podMapper.Process(metrics, mockDeviceInfo)
+	err := podMapper.Process(metrics, mockDeviceInfo)
 	require.NoError(t, err)
 
 	// Verify that labels were added and sanitized correctly
@@ -820,11 +812,7 @@ func TestProcessPodMapper_WithUID(t *testing.T) {
 	}
 
 	// Process metrics
-	// Update cache manually to populate deviceToPod mapping
-	err := podMapper.updateCache(mockDeviceInfo)
-	require.NoError(t, err)
-
-	err = podMapper.Process(metrics, mockDeviceInfo)
+	err := podMapper.Process(metrics, mockDeviceInfo)
 	require.NoError(t, err)
 
 	// Verify that UIDs were added correctly
@@ -935,11 +923,7 @@ func TestProcessPodMapper_WithLabelsAndUID(t *testing.T) {
 	}
 
 	// Process metrics
-	// Update cache manually to populate deviceToPod mapping
-	err := podMapper.updateCache(mockDeviceInfo)
-	require.NoError(t, err)
-
-	err = podMapper.Process(metrics, mockDeviceInfo)
+	err := podMapper.Process(metrics, mockDeviceInfo)
 	require.NoError(t, err)
 
 	// Verify that both labels and UIDs were processed correctly

